src/Featurizers/StandardScaleWrapperFeaturizer.h

```diff
@@ -0,0 +1,348 @@
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class StandardScaleWrapperEstimatorImpl : public TransformerEstimator<InputT, TransformedT> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = TransformerEstimator<InputT, TransformedT>;
+    using TransformerType                   = StandardScalerTransformer<InputT, TransformedT>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    StandardScaleWrapperEstimatorImpl(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex, bool with_mean, bool with_std);
+    ~StandardScaleWrapperEstimatorImpl(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(StandardScaleWrapperEstimatorImpl);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+    bool          const                     _with_mean;
+    bool          const                     _with_std;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+    FitResult fit_impl(typename BaseType::InputType const *, size_t) override;
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the definition is separate from the declaration
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+        // ----------------------------------------------------------------------
+        using StandardStatisticalAnnotationData = Components::StandardStatisticalAnnotationData<InputT>;
+        using StatisticalMetricsEstimator       = Components::StatisticalMetricsEstimator<InputT, MaxNumTrainingItemsV>;
+
+        using StandardDeviationAnnotationData   = Components::StandardDeviationAnnotationData;
+        using StandardDeviationEstimator        = Components::StandardDeviationEstimator<InputT, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+        // if _with_mean is false, average is set to 0
+        std::double_t average = 0;
+        // if _with_std is false, deviation is set to 1
+        std::double_t deviation = 1;
+        StandardStatisticalAnnotationData const &        average_stats(StatisticalMetricsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::StatisticalMetricsEstimatorName));
```

Comment:
These values should only be retrieved if the corresponding flags are set to true.